test/src/e2e_vm_tests/test_programs/should_pass/stdlib/result/src/main.sw

```diff
@@ -35,3 +36,12 @@ fn test_unwrap_ok() {
         revert(0);
     }
 }
+
+fn test_unwrap_or() {
```

Comment:
Do you mind expanding this whole test for `Result` into a more elaborate one similarly to what we did for `Option` here: https://github.com/FuelLabs/sway/tree/master/test/src/e2e_vm_tests/test_programs/should_pass/stdlib/option?